### DISCUSSION TITLE
cmsis-pack-manager

### DISCUSSION THREAD

**Discussion by 0Grit** [Open-CMSIS-Pack/Open-CMSIS-Pack-Spec#40] [General]
Any particular reason that  https://github.com/pyocd/cmsis-pack-manager is not being used as a starting point to manage open-cmsis-packs?

I'd much prefer Rust, and there probably would be an opportunity for the open-cmsis-pack project to benefit from [Cargo](https://doc.rust-lang.org/cargo/) as a package manager.

**Comment by 0Grit** [comment]
@Yatekii Do you have any comments on this topic? I have a feeling work being undertaken by this project could significantly impact your interests.

**Reply by Yatekii** [reply]
In my oppinion, if we make a new spec of this it has to be completely different. XML is not really the thing to go for. Not for the general description and definitely not for the debug sequences.

Generally the cmsis-pack-manager does its job. I am not sure if we would not be better off starting with a completely new format and writing it from scratch.

I do not really get any wiser what the exact purpose of this project here is imo. But we have created our own yaml files for two reasons: a) we want support for other architectures than ARM and b) the packs are VERY flaky in quality and oftentimes bloated in content.

I would welcome a spec that really just describes the target instead of focusing on C templating.

These are my two cents on the matter :) Generally the Rust community would be very interested in a good standard format to describe targets. I even thought about hosting our own repository with device data with focus on actual describing all targets correctly.
